src/h5web/nexus-visualizer/models.ts

```diff
@@ -1,16 +0,0 @@
-import { Vis } from '../dataset-visualizer/models';
```

Comment:
> Moved into `visualizations/nexus`.
> 
> I did remove `NXDataAttribute` as I felt it was an unnecessary abstraction for now -- `getAttributeValue(dataset, 'signal')` is just easier to read in my opinion.

:crying_cat_face: 

I get your point but I think it is better to have a single place where we define Nexus keywords. Moreover, `axes` and `default` will be added to the list in next PRs.
> 
> I also removed `INTERPRETATION_VIS` because I no longer need the mapping. I use the interpretation in the `supportsEntity` functions of the two NeXus visualizations, so I can just check `interpretation === Vis.NxImage` for instance.

:+1: for the rest

Reply:
How about typing `getAttributeValue` so it only accepts a union of keywords, i.e. `type NxAttribute = 'signal' | 'interpretation' | 'axes' | ...` ?

Reply:
I find the `enum` handy as autocompletion ensures that no typos are made when calling the function.

But your proposition is fine as well. I just want to have a structure that defines the keywords.

Reply:
Cool, but note that the union type would also be auto-completed.